app/controllers/mixins/actions/host_actions/misc.rb

```diff
@@ -34,9 +34,8 @@ def process_hosts_refresh(hosts, task, display_name)
 
         def process_hosts_destroy(hosts, display_name)
           each_host(hosts, display_name) do |host|
-            validation = host.validate_destroy
-            if !validation[:available]
-              add_flash(validation[:message], :error)
+            if !host.supports?(:destroy)
+              add_flash(host.unsupported_reason(:destroy), :error)
```

Comment:
@Fryguy so, if the model has N_() for the unsupported reason, then all the UI callers have to translate that via _() in places like this.  Is that correct?  Asking because of [this discussion](https://github.com/ManageIQ/manageiq/pull/21046#pullrequestreview-598001159) in a different PR.

Reply:
Yes, I believe so.  From what I can tell, N_() just tells the i18n engine to consider that string for translation (i.e. put it in the catalog), but don't actually translate it.  That's why it's used for things like constants and enums.  However you have to call _() to actually translate it.  If you were to use _() on a constant it would translate once at constant resolution time, then never translate again for the user.